File: mybatis-flex-test/mybatis-flex-spring-test/src/main/java/com/mybatisflex/test/model/Account.java
/*
 *  Copyright (c) 2022-2023, Mybatis-Flex (fuhai999@gmail.com).
 *  <p>
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *  <p>
 *  http://www.apache.org/licenses/LICENSE-2.0
 *  <p>
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.mybatisflex.test.model;

import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.Table;

import java.util.Date;

@Table(value = "tb_account", onSet = AccountOnSetListener.class)
public class Account {

    @Id
    private Long id;
    private String userName;
    private Integer age;
    private Date birthday;
    private Sex sex;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public Date getBirthday() {
        return birthday;
    }

    public void setBirthday(Date birthday) {
        this.birthday = birthday;
    }

    public Sex getSex() {
        return sex;
    }

    public void setSex(Sex sex) {
        this.sex = sex;
    }
    @Override
    public String toString() {
        return "Account{" +
            "id=" + id +
            ", userName='" + userName + '\'' +
            ", age=" + age +
            ", birthday=" + birthday +
            ", sex=" + sex +
            '}';
    }

}

File: mybatis-flex-kotlin/src/test/java/com/mybatisflex/kotlintest/entry/Account.kt
package com.mybatisflex.kotlintest.entry

import com.mybatisflex.annotation.Column
import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.NoneListener
import com.mybatisflex.annotation.Table
import com.mybatisflex.kotlin.entry.Entry
import java.util.*

/**
 * 测试用数据类（最好不要写成data class，否则需要与数据库字段数据顺序一致）
 * @author 卡莫sama(yuanjiashuai)
 * @date 2023/8/7
 */
@Table(value = "tb_account", onUpdate = [NoneListener::class])
data class Account(
    @Id var id: Int?,
    @Column("u_name") var userName: String?,
    var age: Int?,
    var birthday: Date?,
    ) : Entry(){
    override fun toString(): String {
        return "Account(id=$id, userName=$userName, birthday=$birthday, age=$age)"
    }
}
